### ISSUE TITLE
WYSIWYG unable to set default value in ui component

### ISSUE THREAD

**Issue by milansimek** [magento/magento2#10048]
### Preconditions

1. Magento v2.1.6
2. 

### Steps to reproduce

1. Create a form uicomponent
2. Create a wysiwyg editor field
3. Add a default value

### Expected result

1. The default value should be used when no value is set. This works properly for other fields like textarea's etc.

### Actual result

1. The default value is not used. Editor is empty

### How to fix
Edit 
magento2/app/code/Magento/Ui/view/base/web/js/form/element/wysiwyg.js

Change value in the defaults object to null. Currently it's ''

**Comment by TomashKhamlai** [comment]
Hello! Thank you for submission.
When I was on step 3, I added
```php
            $fieldset->addField(
                'content',
                'editor',
                [
                    'name' => 'content',
                    'label'     => 'WWG',
                    'title'     => 'WYSIWYG',
                    'style'     => 'height:15em',
                    'config'    => $this->_wysiwyg->getConfig(),
                    'wysiwyg'   => true,
                    'required'  => false,
                    'value'     => 'Unusualy default value!'
                ]
            );
```

This worked for me on  Magento v2.2.0. Please provide more details.

**Comment by milansimek** [comment]
Hi, thanks for your reply.

You should create a form ui component (XML), not a form PHP block.

**Comment by TomashKhamlai** [comment]
@milansimek, I interested in reproducing this issue. It would save me some time on reproducing this issue if you could provide me the code for those 3 steps that you described. Please provide the code. Thank you in advance.

**Comment by milansimek** [comment]
@TomashKhamlai  Hi, sorry for my late reply. I missed your comment somehow.

You can use XML code such as the following to add a WYSIWYG editor to a form UI component:

```xml
 <field name="detail">
            <argument name="data" xsi:type="array">
                <item name="config" xsi:type="array">
                    <item name="dataType" xsi:type="string">text</item>
                    <item name="formElement" xsi:type="string">wysiwyg</item>
                    <item name="source" xsi:type="string">sample_shop</item>
                    <item name="label" xsi:type="string">Detail</item>

                    <item name="template" xsi:type="string">ui/form/field</item>
                    <item name="wysiwyg" xsi:type="boolean">true</item>
                    <item name="dataScope" xsi:type="string">detail</item>
                    <item name="sortOrder" xsi:type="number">50</item>
                    <item name="rows" xsi:type="number">8</item>
                     <item name="default" xsi:type="string">this is the default text</item>
                </item>
            </argument>
        </field>
```
The `<item name="default" xsi:type="string">this is the default text</item>` part doesn't work.

If you update `app/code/Magento/Ui/view/base/web/js/form/element/wysiwyg.js` like I mentioned in my first post, it will work properly.

**Comment by magento-engcom-team** [comment]
@milansimek  To set default value into wysiwyg content area, Your have to use "value" instead of "default". For instance, try to change Your code to something like:
```
<field name="detail">
 <argument name="data" xsi:type="array">
  <item name="config" xsi:type="array">
   <item name="dataType" xsi:type="string">text</item>
   <item name="formElement" xsi:type="string">wysiwyg</item>
   <item name="source" xsi:type="string">sample_shop</item>
   <item name="label" xsi:type="string">Detail</item>

   <item name="template" xsi:type="string">ui/form/field</item>
   <item name="wysiwyg" xsi:type="boolean">true</item>
   <item name="dataScope" xsi:type="string">detail</item>
   <item name="sortOrder" xsi:type="number">50</item>
   <item name="rows" xsi:type="number">8</item>
   <item name="value" xsi:type="string">this is the default text</item>
  </item>
 </argument>
</field>
```

**Comment by milansimek** [comment]
@magento-engcom-team While that seems to fix the issue at first glance, it doesn't work correctly. When adding the default value to the "value" element, it will be restored if you save the form field with empty contents.

So for example, set the "value" to "this is the default text".

- Open new form: default value "this is the default text" is set as expected
- Remove the default value from the wysiwyg editor and save the form
- Expected result: the wysiwyg editor should be empty
- Actual result: value is again "this is the default text"

**Comment by magento-engcom-team** [comment]
@milansimek thank you for your bug report.
We've created internal ticket MAGETWO-75440 to track progress on the issue.

**Comment by manuelson** [comment]
Hello,
can i get this issue? 
Thanks

**Comment by okorshenko** [comment]
@manuelson  Sure, please, accept the invitation on the GitHub

**Comment by manuelson** [comment]
@okorshenko done :)

**Comment by prakashsaran** [comment]
hi guys 
anybody tell me how to magento 2 admin  in create text field with default value ?

i try with following code but its doesn't work
$fieldset->addField(
		'current_admin',
		'hidden',
		['name' => 'current_admin', 'values' => $current_admin]
		);

**Comment by sidolov** [comment]
Hi @milansimek. Thank you for your report. 
The issue has been fixed in magento/magento2#19048 by @omiroshnichenko in 2.3-develop branch
Related commit(s):
 - [70743fa4f5e5a6c6ba416d33cfb9d36faaaa47c2](https://github.com/magento/magento2/commit/70743fa4f5e5a6c6ba416d33cfb9d36faaaa47c2)

The fix will be available with the upcoming 2.3.1 release.
